p2p/chain.py

```diff
@@ -82,12 +82,23 @@ def register_peer(self, peer: BasePeer) -> None:
                 self.logger.debug(
                     "Read %s msg from %s's queue; %d msgs pending", cmd, peer, pending_msgs)
 
-            try:
-                await self.handle_msg(peer, cmd, msg)
-            except Exception:
-                self.logger.error("Unexpected error when processing msg from %s: %s",
-                                  peer, traceback.format_exc())
-                break
+            # Our handle_msg() method runs cpu-intensive tasks in sub-processes so that the main
+            # loop can keep processing msgs, and that's why we use ensure_future() instead of
+            # awaiting for it to finish here.
+            asyncio.ensure_future(self.handle_msg(peer, cmd, msg))
+
+    async def handle_msg(self, peer: ETHPeer, cmd: protocol.Command,
+                         msg: protocol._DecodedMsgType) -> None:
+        try:
+            await self._handle_msg(peer, cmd, msg)
+        except OperationCancelled:
+            # Silently swallow OperationCancelled exceptions because we run unsupervised (i.e.
+            # with ensure_future()). Our caller will also get an OperationCancelled anyway, and
+            # there it will be handled.
+            pass
+        except Exception:
+            self.logger.error("Unexpected error when processing msg from %s: %s",
```

Comment:
I think that you can use `self.logger.exception("only the message")` and it will include a stacktrace in the logging output, allowing us to skip the `traceback.formac_exc()` logic.